### PR TITLE
(feat): Add Vaporwave Finance to Zapper.fi

### PR DESCRIPTION
**Pull request by VaporwaveFinance**
## Description

Hello! 

Vaporwave Finance is a yield optimizer on Aurora. This is our integration to Zapper.

## Checklist

- [ x] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] (optional) As a contributor, my Ethereum address/ENS is:
- [ ] (optional) As a contributor, my Twitter handle is:

## How to test?

Add vaporware-finance to enabled apps in the .env file

.env -> ENABLED_APPS=vaporwave-finance

Deploy locally and use the following endpoints:

http://localhost:5001/apps/vaporwave-finance/tokens?groupIds[]=vault&network=aurora
http://localhost:5001/apps/vaporwave-finance/positions?groupIds[]=farm&network=aurora

Retrieved data can be matched to our own API endpoints:

Our list of vaults:
https://api.vaporwave.farm/vaults
Vault token prices:
https://api.vaporwave.farm/vaporTokenPrices
Vault APYs:
https://api.vaporwave.farm/apy
Token prices:
https://api.vaporwave.farm/prices
LP token prices:
https://api.vaporwave.farm/lps

And of course, our app lists this data as well:
www.vaporwave.farm 

We look forward to hearing back from you with anything we need change. Thanks for your time.

### CODE REVIEW

**Comment by immasandwich** [comment]
Apologies for the delays, we'll get to this by tomorrow!

File: src/apps/vaporwave-finance/aurora/vaporwave-finance.farm.contract-position-fetcher.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  ) { }
+
+  async getPositions() {
+    // http://localhost:5001/apps/vaporwave-finance/positions?groupIds[]=farm&network=aurora
+    const wantPrices = await Axios.get("https://api.vaporwave.farm/lps").then(
+      (v) => v.data
+    )
+    const baseTokenPrices = await Axios.get("https://api.vaporwave.farm/prices").then(
+      (v) => v.data
+    )
```
Comment (immasandwich) [inline src/apps/vaporwave-finance/aurora/vaporwave-finance.farm.contract-position-fetcher.ts:39]: You should use `@CacheOnInterval` to reduce calls to your API, see https://studio.zapper.fi/docs/recipes/caching-patterns for details.

File: src/apps/vaporwave-finance/aurora/vaporwave-finance.vault.token-fetcher.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    );
+    const apyData = await Axios.get("https://api.vaporwave.farm/apy").then(
+      (v) => v.data
+    )
+    const wantPrices = await Axios.get("https://api.vaporwave.farm/lps").then(
+      (v) => v.data
+    )
+    const baseTokenPrices = await Axios.get("https://api.vaporwave.farm/prices").then(
+      (v) => v.data
+    )
```
Comment (immasandwich) [inline src/apps/vaporwave-finance/aurora/vaporwave-finance.vault.token-fetcher.ts:96]: Don't use prices for the Vaporwave API. This makes it such that the prices for tokens end up not reflecting other prices in our application. Instead, use the Zapper prices as follows:

```ts
// Get base token prices (USDT, VWAVE, etc.)
const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
// Get dependency app token prices (i.e: Trisolaris, Rose, etc... we're missing these)
const appTokens = await this.appToolkit.getAppTokenPositions(...dependencies);

// ...

// Find the match
const wantToken = [...appTokens, ...baseTokens].find(v => v.address === vault.tokenAddress)
```

**Review by immasandwich — APPROVED** [review]

**Comment by SamIam-0x** [comment]
Hello @VaporwaveFinance !

Looks like you did not provide an address when submitting your PR. I recognize it may have seemed kind of random that we were asking for a wallet address with a PR submission, but we are actually looking for ways to reward Studio contributors in the future. 

Is there an address you’d want to provide? Could even be an empty address, if you do not have an doxxed address

Let me know! Also, if you’d like to privately provide it, feel free to shoot me a TG https://t.me/SamIAm_0x or DM on the Zapper Discord.

Thanks!
Sam

**Comment by SamIam-0x** [comment]
Gentle ping on the above @VaporwaveFinance !

**Pull request merged**